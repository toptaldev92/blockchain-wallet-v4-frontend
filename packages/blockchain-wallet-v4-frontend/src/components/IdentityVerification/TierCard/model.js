import { path, prop } from 'ramda'

export const TIERS = {
  1: {
    time: '3',
    limit: 'ANNUAL',
    requirements: [
      {
        name: 'EMAIL',
        complete: ({ emailVerified }) => emailVerified === 1
      },
      {
        name: 'NAME',
        complete: ({ userData }) =>
          prop('firstName', userData) && prop('lastName', userData)
      },
      {
        name: 'DOB',
        complete: ({ userData }) => prop('dob', userData)
      },
      {
        name: 'ADDRESS',
        complete: ({ userData }) =>
          path(['address', 'city'], userData) &&
          path(['address', 'country'], userData) &&
          path(['address', 'line1'], userData)
      }
    ]
  },
  2: {
    time: '10',
    limit: 'DAILY',
    requirements: [
      {
        name: 'TIER1',
        complete: ({ userTiers }) => userTiers[1].state === 'verified'
      },
      {
        name: 'MOBILE',
        complete: ({ mobileVerified }) => mobileVerified === 1
      },
      {
        name: 'GOVID',
        complete: ({ userData }) => prop('kycState', userData) !== 'NONE'
      },
      {
        name: 'SELFIE',
        complete: ({ userData }) => prop('kycState', userData) !== 'NONE'
      }
    ]
  }
}
