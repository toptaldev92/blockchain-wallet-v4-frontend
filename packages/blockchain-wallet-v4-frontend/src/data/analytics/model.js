export const EXCHANGE = {
  ENTERED: ['exchange', 'checkout', 'entered'],
  FIRST_STEP_SUBMIT: ['exchange', 'checkout', 'first_step_submit'],
  SECOND_STEP_BACK: ['exchange', 'checkout', 'second_step_back'],
  SECOND_STEP_SUBMIT_CLICK: [
    'exchange',
    'checkout',
    'second_step_submit_click'
  ],
  SECOND_STEP_SUBMIT: ['exchange', 'checkout', 'second_step_submit'],
  SECOND_STEP_ERROR: ['exchange', 'checkout', 'second_step_error']
}

export const KYC = {
  EMAIL_EXISTS: ['kyc', 'onboard', 'email_exists'],
  STARTED: ['kyc', 'onboard', 'started_kyc_flow'],
  REENTERED: ['kyc', 'onboard', 'reentered_kyc_flow'],
  PERSONAL_STEP_COMPLETE: ['kyc', 'onboard', 'personal_step_complete_kyc_flow'],
  MOBILE_STEP_COMPLETE: ['kyc', 'onboard', 'mobile_step_complete_kyc_flow'],
  ONFIDO_STARTED: ['kyc', 'onboard', 'onfido_started'],
  COMPLETE: ['kyc', 'onboard', 'complete_kyc_flow']
}

export const TRACKING_ACTIONS = ['click', 'change', 'timing']
