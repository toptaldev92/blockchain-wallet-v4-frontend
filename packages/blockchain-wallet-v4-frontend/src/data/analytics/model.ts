//
// CUSTOM VARIABLES
//
export const CUSTOM_VARIABLES = {
  KYC_TIER: {
    ID: 1,
    NAME: 'kyc_tier'
  },
  CRYPTO_BALANCES: {
    ID: 2,
    NAME: 'crypto_balances'
  }
}

//
// AB TESTS
//
export enum AB_TESTS {
  VERIFY_EMAIL = 'VerifyEmail'
}

//
// EVENTS
// format: [event_category, event_action, ?event_name, ?event_value]
//

export const ADS_EVENTS = {
  CLICK_AD: ['navigation', 'click_ad'],
  CLICK_AD_TAX: ['navigation', 'click_ad_tax']
}

export const AB_TEST_EVENTS = {
  DISMISS_VERIFICATION: ['email_ab_test', 'dismiss_verification'],
  EMAIL_VERIFIED: ['email_ab_test', 'email_verified']
}

export const PREFERENCE_EVENTS = {
  GENERAL: {
    ENABLE_BTC_LINKS: 'enable_btc_links'
  },
  SECURITY: {
    VERIFY_DEVICE_ACCEPTED: [
      'preferences',
      'security',
      'verify_device_accepted'
    ],
    VERIFY_DEVICE_EMAIL_SENT: [
      'preferences',
      'security',
      'verify_device_email_sent'
    ],
    VERIFY_DEVICE_REJECTED: [
      'preferences',
      'security',
      'verify_device_rejected'
    ]
  }
}
export const SWAP_EVENTS = {
  ORDER_PREVIEW_ERROR: ['swap', 'order_form', 'order_preview_error'],
  USE_MIN: ['swap', 'order_form', 'use_min'],
  USE_MAX: ['swap', 'order_form', 'use_max']
}
export const TRANSACTION_EVENTS = {
  SEND: ['transactions', 'send'],
  REQUEST: ['transactions', 'request'],
  EDIT_DESCRIPTION: ['transactions', 'edit_description'],
  PAYMENT_REQUEST: ['transactions', 'payment_request'],
  SEND_FAILURE: ['send_failure'],
  VIEW_TX_ON_EXPLORER: ['transactions', 'view_tx_explorer'],
  BITPAY_URL_DEEPLINK: ['transactions', 'bitpay', 'bitpay_url_deeplink'],
  BITPAY_FAILURE: ['transactions', 'bitpay', 'bitpay_payment_failure'],
  BITPAY_SUCCESS: ['transactions', 'bitpay', 'bitpay_payment_success']
}

export const EXCHANGE_EVENTS = {
  BANNER_GET_STARTED: ['exchange', 'homepage', 'homepage_banner_click'],
  BUY_SELL_LINKOUT_CLICKED: ['exchange', 'buy_sell_page', 'linkout_click'],
  BUY_SELL_CONNECT_WALLET_CLICKED: [
    'exchange',
    'buy_sell_page',
    'connect_wallet_click'
  ],
  CONNECT_NOW: ['exchange', 'link_page', 'connect_now_click'],
  LEARN_MORE: ['exchange', 'link_page', 'learn_more_click'],
  LINKED_WALLET_LINKOUT_CLICKED: [
    'exchange',
    'homepage',
    'linked_wallet_linkout_clicked'
  ],
  PROMO: ['exchange', 'send_modal']
}

export const PRICE_CHART_EVENTS = {
  CLICK_BUY_BITCOIN: ['price_chart', 'footer_button', 'buy_bitcoin'],
  CLICK_SWAP_COIN: ['price_chart', 'footer_button', 'swap_coin']
}

export const SB_EVENTS = {
  CLICK_SUPPORT_ARTICLE: ['simple_buy', 'transfer_flyout', 'read_it_now']
}

export const INTEREST_EVENTS = {
  HOME: {
    CLICK_SUPPORT_ARTICLE: ['interest', 'home', 'learn_more_click'],
    VIEW_TX_SUMMARY: ['interest', 'home', 'view_tx_click']
  },
  DEPOSIT: {
    SEND_SUCCESS: ['interest', 'deposit', 'send_success'],
    SEND_FAILURE: ['interest', 'deposit', 'send_failure']
  },
  WITHDRAWAL: {
    REQUEST_SUCCESS: ['interest', 'withdrawal', 'request_success'],
    REQUEST_FAILURE: ['interest', 'withdrawal', 'request_failure']
  }
}
