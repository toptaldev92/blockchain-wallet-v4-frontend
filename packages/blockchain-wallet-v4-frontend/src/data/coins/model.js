import { filter, keys } from 'ramda'

// TODO list:
// combine with or put this into wallet options
// update home page charts component to use this model
// move this model to core package and add the following data to model
//      blockchain-wallet-v4/src/redux/data/eth/model.js
//      blockchain-wallet-v4/src/exchange/currencies/PAX.js
//      blockchain-wallet-v4/src/redux/kvStore/eth/model.js

// COIN MODEL OVERVIEW
/*
 * name:              [REQUIRED] real name of coin/token (hidden from user)
 * displayName:       [REQUIRED] display name of coin/token (shown to user)
 * coinCode:          [REQUIRED] real coin code (hidden from user)
 * coinCodeDisplay:   [REQUIRED] display coin code (shown to user)
 * txExplorerBaseUrl: [REQUIRED] base url to coin/token tx explorer
 * icons:             [REQUIRED] map of the various icons for coin/token
 * minConfirmations:  [REQUIRED] confirmations needed for tx
 * isErc20:           [REQUIRED] bool indicating if the coin is an ERC20 token
 * hasLockboxSupport: [REQUIRED] bool indicating if the coin is available on lockbox
 * colorCode:         [REQUIRED] primary color of coin/token used for icons and text
 * txListAppRoute:    [OPTIONAL] app route for coin/token tx list page
 * learnMoreLink:     [OPTIONAL] external url explaining coin/token
 * campaign:          [OPTIONAL] string indicating active campaign for coin/token
 * airdrop:           [OPTIONAL] metadata about current campaign/airdrop
 * showNewTagSidenav: [OPTIONAL] show new tag in sidenav for coin/token (coin must be inSIDENAV_COIN_ORDER const)
 */
export const COIN_MODELS = {
  BTC: {
    campaign: null,
    coinCode: 'BTC',
    coinCodeDisplay: 'BTC',
    colorCode: 'btc',
    displayName: 'Bitcoin',
    hasLockboxSupport: true,
    icons: {
      default: 'btc',
      circle: 'btc-circle',
      circleFilled: 'btc-circle-filled'
    },
    isErc20: false,
    learnMoreLink: 'https://www.blockchain.com/learning-portal/bitcoin-faq',
    minConfirmations: 3,
    name: 'Bitcoin',
    txExplorerBaseUrl: 'https://blockchain.com/btc/tx',
    txListAppRoute: '/btc/transactions',
    showNewTagSidenav: false
  },
  BCH: {
    campaign: null,
    coinCode: 'BCH',
    coinCodeDisplay: 'BCH',
    colorCode: 'bch',
    displayName: 'Bitcoin Cash',
    hasLockboxSupport: true,
    icons: {
      default: 'bch',
      circle: 'bch-circle',
      circleFilled: 'bch-circle-filled'
    },
    isErc20: false,
    learnMoreLink: null,
    minConfirmations: 3,
    name: 'Bitcoin Cash',
    txExplorerBaseUrl: 'https://www.blockchain.com/bch/tx',
    txListAppRoute: '/bch/transactions',
    showNewTagSidenav: false
  },
  BSV: {
    campaign: null,
    coinCode: 'BSV',
    coinCodeDisplay: 'BSV',
    colorCode: 'bsv',
    displayName: 'Bitcoin SV',
    hasLockboxSupport: false,
    icons: {
      default: 'bsv'
    },
    isErc20: false,
    learnMoreLink: null,
    minConfirmations: 3,
    name: 'Bitcoin SV',
    txExplorerBaseUrl: 'https://blockchair.com/bitcoin-sv/transaction',
    txListAppRoute: null,
    showNewTagSidenav: false
  },
  ETH: {
    campaign: null,
    coinCode: 'ETH',
    coinCodeDisplay: 'ETH',
    colorCode: 'eth',
    displayName: 'Ethereum',
    hasLockboxSupport: true,
    icons: {
      default: 'eth',
      circle: 'eth-circle',
      circleFilled: 'eth-circle-filled'
    },
    isErc20: false,
    learnMoreLink: 'https://www.blockchain.com/learning-portal/ether-basics',
    minConfirmations: 12,
    name: 'Ethereum',
    txExplorerBaseUrl: 'https://www.blockchain.com/eth/tx',
    txListAppRoute: '/eth/transactions',
    showNewTagSidenav: false
  },
  PAX: {
    campaign: null,
    coinCode: 'PAX',
    coinCodeDisplay: 'USDp',
    colorCode: 'pax',
    displayName: 'USD Pax',
    hasLockboxSupport: false,
    icons: {
      default: 'usdp',
      circle: 'usdp',
      circleFilled: 'usdp'
    },
    isErc20: true,
    learnMoreLink: null,
    minConfirmations: 12,
    name: 'Paxos',
    txExplorerBaseUrl: 'https://www.blockchain.com/eth/tx',
    txListAppRoute: '/pax/transactions',
    showNewTagSidenav: true
  },
  XLM: {
    airdrop: {
      name: 'SUNRIVER',
      link:
        'https://support.blockchain.com/hc/en-us/categories/360001126692-Crypto-Giveaway',
      image: 'stellar-planet'
    },
    colorCode: 'xlm',
    campaign: 'sunriver',
    coinCode: 'XLM',
    coinCodeDisplay: 'XLM',
    displayName: 'Stellar',
    hasLockboxSupport: true,
    icons: {
      default: 'xlm',
      circle: 'xlm-circle',
      circleFilled: 'xlm-circle-filled'
    },
    isErc20: false,
    learnMoreLink: '',
    minConfirmations: 1,
    name: 'Stellar',
    txExplorerBaseUrl: 'https://stellarchain.io/tx',
    txListAppRoute: '/xlm/transactions',
    showNewTagSidenav: false
  }
}

// list of supported ERC 20 tokens
export const ERC20_COIN_LIST = keys(filter(c => c.isErc20, COIN_MODELS))
