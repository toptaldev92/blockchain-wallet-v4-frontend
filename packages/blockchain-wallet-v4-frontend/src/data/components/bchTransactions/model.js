export const FORM = '@TX.BCH.FORM'
