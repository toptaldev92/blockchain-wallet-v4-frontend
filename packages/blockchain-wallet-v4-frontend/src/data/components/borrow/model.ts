import { CoinType, LoanType, OfferType } from 'core/types'
import { Exchange } from 'blockchain-wallet-v4/src'

export const INVALID_COIN_TYPE = 'Invalid coin type'

export const NO_OFFER_EXISTS = 'NO_OFFER_EXISTS'

export const BORROW_STEPS = {
  CHECKOUT: 0,
  DETAILS: 1,
  ADD_COLLATERAL: 2
}

export const getCollateralizationColor = (
  displayName: 'safe' | 'risky' | 'unsafe'
) => {
  switch (displayName) {
    case 'safe':
      return 'green600'
    case 'risky':
      return 'orange600'
    case 'unsafe':
      return 'red600'
  }
}

export const getCollateralAmtRequired = (loan: LoanType, offer: OfferType) => {
  return (
    (offer.terms.collateralRatio - loan.collateralisationRatio) *
    Number(loan.principal.amount[0].value)
  )
}

export const getCollateralizationDisplayName = (
  currentCollateral: number,
  offer: OfferType
) => {
  switch (true) {
    case currentCollateral <= offer.callTerms.liquidationHardRatio:
      return 'unsafe'
    case currentCollateral <= offer.callTerms.callTriggerRatio:
      return 'risky'
    default:
      return 'safe'
  }
}

export const fiatDisplayName = (coin: CoinType) => {
  switch (coin) {
    case 'PAX':
      return 'USD'
    default:
      return 'USD'
  }
}

export const getAmount = (value: number, coin: CoinType) => {
  switch (coin) {
    case 'BTC':
      return Exchange.convertBtcToBtc({ value, fromUnit: 'BTC', toUnit: 'SAT' })
        .value
  }
}
