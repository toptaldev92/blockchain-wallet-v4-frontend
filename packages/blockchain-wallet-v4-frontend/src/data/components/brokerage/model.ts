import { WalletFiatType } from 'blockchain-wallet-v4/src/types'

export const DEFAULT_METHODS = {
  currency: 'EUR' as WalletFiatType,
  methods: []
}
