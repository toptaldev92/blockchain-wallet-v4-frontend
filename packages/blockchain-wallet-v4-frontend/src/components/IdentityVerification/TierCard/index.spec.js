import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { TierCard } from './index'
import { INITIAL_TIERS } from 'data/modules/profile/model.js'

const userTiers = INITIAL_TIERS

describe('TierCard', () => {
  it('renders null w/o showBanner', () => {
    const component = shallow(<TierCard tier={1} userTiers={userTiers} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with column view', () => {
    const component = shallow(
      <TierCard tier={2} userTiers={userTiers} column />
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
