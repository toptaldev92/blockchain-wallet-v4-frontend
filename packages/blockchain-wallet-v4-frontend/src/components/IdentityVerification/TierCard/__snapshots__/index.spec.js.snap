// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TierCard renders correctly with column view 1`] = `
<TierCard__Wrapper
  className=" column"
>
  <TierCard__Announcement
    color="white"
    size="18px"
    uppercase={true}
    weight={500}
  >
    <FormattedMessage
      defaultMessage="Get Free Crypto"
      id="components.identityverification.tiercard.getfreecrypto"
      values={Object {}}
    />
  </TierCard__Announcement>
  <TierCard__Container>
    <TierCard__Header
      color="marketing-primary"
      uppercase={true}
    >
      <FormattedMessage
        defaultMessage="Tier {tier} Verification"
        id="components.identityverification.tiercard.tierheader"
        values={
          Object {
            "tier": 2,
          }
        }
      />
    </TierCard__Header>
    <TierCard__Content>
      <TierCard__Row>
        <TierCard__Column>
          <Text
            altFont={false}
            capitalize={false}
            color="marketing-secondary"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="32px"
            uppercase={false}
            weight={400}
          >
            $25,000
          </Text>
          <Text
            altFont={false}
            capitalize={false}
            color="textBlack"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="14px"
            style={
              Object {
                "marginTop": "8px",
              }
            }
            uppercase={false}
            weight={400}
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="14px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Daily Swap Limit"
                id="components.identityverification.tiercard.dailylimit"
                values={Object {}}
              />
            </Text>
          </Text>
          <Text
            altFont={false}
            capitalize={false}
            color="gray-3"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="14px"
            style={
              Object {
                "marginTop": "7px",
              }
            }
            uppercase={false}
            weight={400}
          >
            <FormattedMessage
              defaultMessage="Takes {time} min"
              id="components.identityverification.tiercard.timing"
              values={
                Object {
                  "time": "10",
                }
              }
            />
          </Text>
        </TierCard__Column>
        <TierCard__Column>
          <TextGroup
            inline={true}
            key="0"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-3"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Tier 1+"
                id="components.identityverification.tiercard.tier1"
                values={Object {}}
              />
            </Text>
          </TextGroup>
          <TextGroup
            inline={true}
            key="1"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Verified Phone"
                id="components.identityverification.tiercard.phone"
                values={Object {}}
              />
            </Text>
          </TextGroup>
          <TextGroup
            inline={true}
            key="2"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Gov't Issued ID"
                id="components.identityverification.tiercard.id"
                values={Object {}}
              />
            </Text>
            <Icon
              color="success"
              cursor={false}
              name="check"
              size="12px"
              style={
                Object {
                  "marginLeft": "5px",
                }
              }
              weight={400}
            />
          </TextGroup>
          <TextGroup
            inline={true}
            key="3"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Selfie"
                id="components.identityverification.tiercard.selfie"
                values={Object {}}
              />
            </Text>
            <Icon
              color="success"
              cursor={false}
              name="check"
              size="12px"
              style={
                Object {
                  "marginLeft": "5px",
                }
              }
              weight={400}
            />
          </TextGroup>
        </TierCard__Column>
      </TierCard__Row>
    </TierCard__Content>
    <TierCard__ActionButton
      fullwidth={true}
      jumbo={true}
      nature="primary"
    >
      <FormattedMessage
        defaultMessage="Get Started"
        id="components.identityverification.tiercard.getstarted"
        values={Object {}}
      />
    </TierCard__ActionButton>
  </TierCard__Container>
</TierCard__Wrapper>
`;

exports[`TierCard renders null w/o showBanner 1`] = `
<TierCard__Wrapper
  className=""
>
  <TierCard__Container>
    <TierCard__Header
      color="marketing-primary"
      uppercase={true}
    >
      <FormattedMessage
        defaultMessage="Tier {tier} Verification"
        id="components.identityverification.tiercard.tierheader"
        values={
          Object {
            "tier": 1,
          }
        }
      />
    </TierCard__Header>
    <TierCard__Content>
      <TierCard__Row>
        <TierCard__Column>
          <Text
            altFont={false}
            capitalize={false}
            color="marketing-secondary"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="32px"
            uppercase={false}
            weight={400}
          >
            $1,000
          </Text>
          <Text
            altFont={false}
            capitalize={false}
            color="textBlack"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="14px"
            style={
              Object {
                "marginTop": "8px",
              }
            }
            uppercase={false}
            weight={400}
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="14px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Annual Swap Limit"
                id="components.identityverification.tiercard.annuallimit"
                values={Object {}}
              />
            </Text>
          </Text>
          <Text
            altFont={false}
            capitalize={false}
            color="gray-3"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="14px"
            style={
              Object {
                "marginTop": "7px",
              }
            }
            uppercase={false}
            weight={400}
          >
            <FormattedMessage
              defaultMessage="Takes {time} min"
              id="components.identityverification.tiercard.timing"
              values={
                Object {
                  "time": "3",
                }
              }
            />
          </Text>
        </TierCard__Column>
        <TierCard__Column>
          <TextGroup
            inline={true}
            key="0"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Verified Email"
                id="components.identityverification.tiercard.email"
                values={Object {}}
              />
            </Text>
          </TextGroup>
          <TextGroup
            inline={true}
            key="1"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Name"
                id="components.identityverification.tiercard.name"
                values={Object {}}
              />
            </Text>
          </TextGroup>
          <TextGroup
            inline={true}
            key="2"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Date of Birth"
                id="components.identityverification.tiercard.dob"
                values={Object {}}
              />
            </Text>
          </TextGroup>
          <TextGroup
            inline={true}
            key="3"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Address"
                id="components.identityverification.tiercard.address"
                values={Object {}}
              />
            </Text>
          </TextGroup>
        </TierCard__Column>
      </TierCard__Row>
    </TierCard__Content>
    <TierCard__ActionButton
      fullwidth={true}
      jumbo={true}
      nature="primary"
    >
      <FormattedMessage
        defaultMessage="Get Started"
        id="components.identityverification.tiercard.getstarted"
        values={Object {}}
      />
    </TierCard__ActionButton>
  </TierCard__Container>
</TierCard__Wrapper>
`;
